backend/server/services/questionService.js

```diff
@@ -1,26 +1,22 @@
 const questionRepository = require('../repositories/questionRepository');
 
-function missingInputsThrowsValidationError(title, complexity, description) {
+const missingInputsThrowsValidationError = (title, complexity, description) => {
   const innerText = description.replace(/<[^>]+>|\s+/g, '');
   if (!title || !complexity || !innerText) {
     throw { status: 400, message: 'Missing inputs' };
   }
-}
+};
 
-async function duplicateTitleThrowsDuplicateError(id, title) {
+const duplicateTitleThrowsDuplicateError = async (id, title) => {
   // id == null then simply check if title exist, else cross-check id
   const isDuplicateTitle = await questionRepository.findByTitle(title);
-  if (isDuplicateTitle) {
-    if (id == null) {
-      throw { status: 409, message: `Question title already exist` };
-    } 
+  if (!isDuplicateTitle) return;
 
-    const isCurrent = await questionRepository.findByIdAndTitle(id, title);
-    if (!isCurrent) {
-      throw { status: 409, message: `Question title already exist` };
-    }
+  // Duplicate found, check if it is the same question
+  if (!isDuplicateTitle._id.equals(id)) {
+    throw { status: 409, message: `Question title already exist` };
   }
-}
+};
```

Comment:
Since now its just 2 `if` conditions, it may be cleaner to just do
`if (isDuplicateTitle && !isDuplicateTitle._id.equals(id)) {...}`
where the first part check if that `isDuplicateTitle` is not `null` and the second part checks for conflict in `id`.

Reply:
Resolved